modules/aws/aws-infra/README.md

```diff
@@ -0,0 +1,422 @@
+
+A comprehensive, production-ready AWS infrastructure module that provides all necessary resources for Databricks workloads using official AWS Terraform modules and best practices.
+
+## Overview
+
+This module creates a complete AWS infrastructure foundation optimized for Databricks, featuring:
+
+- **🔧 Simplified Configuration**: Uses official `terraform-aws-modules/vpc` for networking
+- **🔒 Secure Storage**: S3 buckets with encryption for workspace and Unity Catalog
+- **👤 IAM Integration**: Cross-account and Unity Catalog roles with Databricks-generated policies
+- **🔗 VPC Endpoints**: Private access to AWS services (S3, STS, Kinesis)
+- **🛡️ Network Firewall**: Configurable FQDN and network-based filtering (optional)
+- **🌐 Hub-Spoke Architecture**: Transit Gateway with centralized internet egress (optional)
+- **🔐 Private Link**: Databricks Private Link endpoints (optional)
+
+## Architecture
+
+### Basic Architecture
+
+```
+┌─────────────────────────────────────────────┐
+│  VPC (10.0.0.0/16)                         │
+│                                             │
+│  ┌──────────────┐  ┌──────────────┐       │
+│  │ Private Sub  │  │ Private Sub  │       │
+│  │ (AZ-a)       │  │ (AZ-b)       │       │
+│  │ Databricks   │  │ Databricks   │       │
+│  └──────┬───────┘  └──────┬───────┘       │
+│         │                  │                │
+│         └─────────┬────────┘                │
+│                   │                         │
+│         ┌─────────▼─────────┐              │
+│         │   NAT Gateway     │              │
+│         │  (Public Subnet)  │              │
+│         └─────────┬─────────┘              │
+│                   │                         │
+│         ┌─────────▼─────────┐              │
+│         │ Internet Gateway  │              │
+│         └───────────────────┘              │
+└─────────────────────────────────────────────┘
+            │
+            ▼
+        Internet
+```
+
+### Hub-Spoke Architecture with Firewall
+
+```
+┌──────────────────────────────────────────────────┐
+│  Spoke VPC (Databricks - 10.0.0.0/16)          │
+│  Private Subnets                                 │
+└────────────────┬─────────────────────────────────┘
+                 │ Transit Gateway
+                 ▼
+┌──────────────────────────────────────────────────┐
+│  Hub VPC (10.1.0.0/16)                          │
+│                                                  │
+│  Private Subnet → Network Firewall → NAT → IGW │
+│  (TGW attach)     (Inspection)                  │
+└──────────────────────────────────────────────────┘
+                 │
+                 ▼
+             Internet
+```
+
+## Module Components
+
+### Core Components (Always Created)
+- **networking.tf** - VPC, subnets, security groups, NAT gateway (via AWS VPC module)
+- **workspacestorage.tf** - Root S3 bucket for Databricks workspace
+- **ucstorage.tf** - Unity Catalog S3 buckets (metastore & data)
```

Comment:
I think that it makes sense to have a separate module for UC S3 buckets, but allow them to be linked with this module.